### PR TITLE
Feature/improve docs

### PR DESCRIPTION
**Pull request by RoaldL**
Improved comments

### CODE REVIEW

File: digital_comms/ccam.py [hunk 1]
```diff
@@ -98,13 +160,35 @@ def population_density(self):
...
+        Arguments
+        ---------
+        pcd_sector: PostcodeSector
+            Representation of a postcode sector that needs to be
+            added to the local area district
+        """
         self._pcd_sectors[pcd_sector.id] = pcd_sector
 
     def add_asset(self, asset):
+        """Add an asset to postcode sector to the local area district
```
Comment (ej550) [inline digital_comms/ccam.py]: Does this not just add an asset to the pcd_sector, and the pcd_sector is added to the LAD in the above method?

Hence, should this not just be """Add an asset to postcode sector"""?

File: digital_comms/ccam.py [hunk 2]
```diff
@@ -98,13 +160,35 @@ def population_density(self):
...
+        Arguments
+        ---------
+        asset: TODO
+            TODO
+        """
         pcd_sector_id = asset.pcd_sector
         self._pcd_sectors[pcd_sector_id].add_asset(asset)
 
     def system(self):
+        """TODO
```
Comment (ej550) [inline digital_comms/ccam.py]: I think this just populates a dict with all of the existing assets, which in total represent the 'system'

File: digital_comms/ccam.py [hunk 3]
```diff
@@ -117,7 +201,16 @@ def system(self):
         return system
 
     def capacity(self):
-        """Return the mean capacity from all nested postcode sectors
+        """Calculate capacity as the mean capacity from all nested postcode sectors
```
Comment (ej550) [inline digital_comms/ccam.py]: Perhaps this should just be 'Calculate mean capacity from all nested postcode sectors' for simplicity

File: digital_comms/ccam.py [hunk 4]
```diff
@@ -128,7 +221,16 @@ def capacity(self):
         return summed_capacity / len(self._pcd_sectors)
 
     def demand(self):
-        """Return the mean capacity demand from all nested postcode sectors
+        """Calculate demand as the mean capacity demand from all nested postcode sectors
```
Comment (ej550) [inline digital_comms/ccam.py]: Should be: """Calculate demand per square kilometer (Mbps km^2) from all nested postcode sectors."""

File: digital_comms/ccam.py [hunk 5]
```diff
@@ -144,7 +246,16 @@ def demand(self):
         return summed_demand / summed_area
 
     def coverage(self):
-        """Return proportion of population with capacity coverage over a threshold
+        """Calculate coverage as the proportion of population with capacity coverage over a threshold
```
Comment (ej550) [inline digital_comms/ccam.py]: """Calculate coverage as the proportion of the population able to obtain the specified capacity threshold"""

File: digital_comms/ccam.py [hunk 6]
```diff
@@ -144,7 +246,16 @@ def demand(self):
...
+
+        Returns
+        -------
+        obj
+            Coverage in the local area district
+
+        Notes
+        -----
+        Function returns `0` when no postcode sectors are configured to the LAD.
```
Comment (ej550) [inline digital_comms/ccam.py]: Are we missing the code for this method?

File: digital_comms/ccam.py [hunk 7]
```diff
@@ -160,7 +271,7 @@ def coverage(self):
 
 
 class PostcodeSector(object):
-    """Represents a pcd_sector to be modelled
+    """""Represents a Postcode sector to be modelled
```
Comment (ej550) [inline digital_comms/ccam.py]: Have we got the doc string wrong here, shouldn't it be three double quotes (""")?

File: digital_comms/ccam.py [hunk 8]
```diff
@@ -223,9 +348,10 @@ def threshold_demand(self, SERVICE_OBLIGATION_CAPACITY):
 
     @property
     def demand(self):
-        """Estimate total demand based on population and penetration
+        """obj: The demand in capacity per km^2
```
Comment (ej550) [inline digital_comms/ccam.py]: My concern is that if this is km^2, why do we then divide by area again in the above object? Need to double check this.

File: digital_comms/ccam.py [hunk 9]
```diff
@@ -223,9 +348,10 @@ def threshold_demand(self, SERVICE_OBLIGATION_CAPACITY):
...
 
-        E.g.
+        Notes
+        -----
             0.02 Mbps per user during busy hours
                 * 100 population
```
Comment (ej550) [inline digital_comms/ccam.py]: I'm not sure I understand this example - double check.

File: digital_comms/ccam.py [hunk 10]
```diff
@@ -239,7 +365,7 @@ def demand(self):
 
     @property
     def population_density(self):
-        """Calculate population density
+        """obj: The population density in people per TODO
```
Comment (ej550) [inline digital_comms/ccam.py]: """obj: Population density in persons per square kilometer (km^2)"""

File: digital_comms/ccam.py [hunk 11]
```diff
@@ -289,15 +415,29 @@ def _small_cell_capacity(self):
 
     @property
     def capacity_margin(self):
+        """obj: The capacity margin in this postcode sector TODO
```
Comment (ej550) [inline digital_comms/ccam.py]: """obj: Capacity margin per postcode sector in Mbps"""

File: digital_comms/ccam.py [hunk 12]
```diff
@@ -289,15 +415,29 @@ def _small_cell_capacity(self):
...
         capacity_margin = self.capacity - self.demand
         return capacity_margin
 
 
 def pairwise(iterable):
     """Return iterable of 2-tuples in a sliding window
 
+    Parameters
+    ----------
+    iterable: list
```
Comment (ej550) [inline digital_comms/ccam.py]: I'm not familiar with pairwise - perhaps you can explain it to me?

File: digital_comms/ccam.py [hunk 13]
```diff
@@ -289,15 +415,29 @@ def _small_cell_capacity(self):
...
+    list of tuple
+        Iterable of 2-tuples
+
+    Example
+    -------
         >>> list(pairwise([1,2,3,4]))
-        [(1,2),(2,3),(3,4)]
+            [(1,2),(2,3),(3,4)]
     """
     a, b = tee(iterable)
```
Comment (ej550) [inline digital_comms/ccam.py]: What does this do?

File: digital_comms/ccam.py [hunk 14]
```diff
@@ -307,10 +447,39 @@ def pairwise(iterable):
...
-    clutter_lookup : list of (population_density_upper_bound, geotype) tuples
-        sorted by population_density_upper_bound ascending
+    Parameters
+    ----------
+    clutter_lookup: list of tuple
+        Lookup table that represents geographical types and their population density
+        sorted by population_density_upper_bound ascending.
+
+        * 0: :obj:`int`
+            Population density in TODO
```
Comment (ej550) [inline digital_comms/ccam.py]: population density should be in persons per square kilometer

File: digital_comms/interventions.py [hunk 15]
```diff
@@ -144,6 +144,31 @@ def decide_interventions(strategy, budget, service_obligation_capacity,
...
+    Returns
+    -------
+    tuple
+        0: `obj`:`list` of `obj`:`dict`  
+            Details of the assets that were built
+
+            Each containing the keys
+
+                site_ngr: str
+                    TODO
```
Comment (ej550) [inline digital_comms/interventions.py]: Unique site reference number

File: digital_comms/ccam.py (hunk 6 above)
Comment (RoaldL) [reply digital_comms/ccam.py]: No I don't think so. Line 261 - 262 implement this behavior.

File: digital_comms/ccam.py (hunk 8 above)
Comment (RoaldL) [reply digital_comms/ccam.py]: Ok I've added a Todo

File: digital_comms/ccam.py (hunk 9 above)
Comment (RoaldL) [reply digital_comms/ccam.py]: This example has been taken over from the original documentation. I will add a todo for the double check

File: digital_comms/ccam.py (hunk 12 above)
Comment (RoaldL) [reply digital_comms/ccam.py]: Me neither when I first looked at this. Therefore I've added an example to the docstring:

    Example
    -------
        >>> list(pairwise([1,2,3,4]))
            [(1,2),(2,3),(3,4)]

File: digital_comms/ccam.py (hunk 13 above)
Comment (RoaldL) [reply digital_comms/ccam.py]: Its just an example that will be added to the documentation

File: digital_comms/ccam.py [hunk 16]
```diff
@@ -83,12 +141,16 @@ def __repr__(self):
...
     def population(self):
+        """obj: Sum of all postcode sectors populations in the local area district
+        """
         return sum([
             pcd_sector.population
             for pcd_sector in self._pcd_sectors.values()])
 
     @property
     def population_density(self):
+        """obj: The average population density in the local area district
```
[the remark this answers is not in the document]
Comment (RoaldL) [reply digital_comms/ccam.py]: Done

File: digital_comms/ccam.py (hunk 1 above)
Comment (RoaldL) [reply digital_comms/ccam.py]: Done

File: digital_comms/ccam.py (hunk 2 above)
Comment (RoaldL) [reply digital_comms/ccam.py]: Done

File: digital_comms/ccam.py (hunk 3 above)
Comment (RoaldL) [reply digital_comms/ccam.py]: Done

File: digital_comms/ccam.py (hunk 4 above)
Comment (RoaldL) [reply digital_comms/ccam.py]: Done

File: digital_comms/ccam.py (hunk 5 above)
Comment (RoaldL) [reply digital_comms/ccam.py]: Done

File: digital_comms/ccam.py (hunk 7 above)
Comment (RoaldL) [reply digital_comms/ccam.py]: Done

File: digital_comms/ccam.py (hunk 10 above)
Comment (RoaldL) [reply digital_comms/ccam.py]: Done

File: digital_comms/ccam.py (hunk 11 above)
Comment (RoaldL) [reply digital_comms/ccam.py]: Done

File: digital_comms/ccam.py (hunk 14 above)
Comment (RoaldL) [reply digital_comms/ccam.py]: Done

File: digital_comms/interventions.py (hunk 15 above)
Comment (RoaldL) [reply digital_comms/interventions.py]: Done

**Pull request merged**